h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoost.java

```diff
@@ -345,14 +347,18 @@ final void buildModel() {
       }
     }
     
-    private XGBoostExecutor makeExecutor(XGBoostModel model) {
-      String propVal = H2O.getSysProperty("xgboost.external.address", null);
-      if (propVal == null) {
-        return new LocalXGBoostExecutor(model, _train);
+    private XGBoostExecutor makeExecutor(XGBoostModel model) throws IOException {
+      if (H2O.ARGS.use_external_xgboost) {
+        return SteamExecutorStarter.getInstance().getRemoteExecutor(model, _train);
       } else {
-        String userName = H2O.getSysProperty("xgboost.external.user", null);
-        String password = H2O.getSysProperty("xgboost.external.password", null);
-        return new RemoteXGBoostExecutor(propVal, model, _train, userName, password);
+        String remoteUriFromProp = H2O.getSysProperty("xgboost.external.address", null);
+        if (remoteUriFromProp == null) {
+          return new LocalXGBoostExecutor(model, _train);
+        } else {
+          String userName = H2O.getSysProperty("xgboost.external.user", null);
```

Comment:
Why is password in system properties? Can this be avoided? Is it passed using `-D` and visible eg. in process list?

Reply:
This is a way to use external cluster used mainly for testing, system properties can also be set via Rapids.